app/common/data/interfaces/user.py

```diff
@@ -29,7 +29,12 @@ def get_platform_admin_users() -> Sequence[User]:
     ).all()
 
 
-def get_or_create_user(email_address: str, name: Optional[str] = None) -> User:
+def get_or_create_user(email_address: str, name: Optional[str] = None, read_only: bool = False) -> Optional[User]:
+    if read_only:
+        try:
+            return db.session.execute(select(User).where(User.email == email_address)).scalar_one()
+        except NoResultFound:
+            return None
```

Comment:
I don't think the interface here feels very sensible - the function is called "get or create user", but then you take an argument that turns off half of that (the ability to create).

Can we just keep this as-is and add a separate "get_user" method that is read-only? Then use the appropriate one in views please.